### PR TITLE
Add example project

### PR DESCRIPTION
**Pull request by mikefowler**
## Overview
Adds an example project that illustrates how to set up
`graphql-sequelize` and `dataloader-sequelize` with `graphql-yoga`.

Closes https://github.com/mickhansen/graphql-sequelize/issues/610

## Reviewers
to: @mickhansen @intellix 
cc: @MauriceMahan

### CODE REVIEW

File: examples/graphql-yoga/src/server.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+const server = new GraphQLServer({
+  typeDefs,
+  resolvers,
+  context(req) {
+    // For each request, create a DataLoader context for Sequelize to use
+    const dataloaderContext = createContext(models.sequelize);
+
+    // Tell `graphql-sequelize` where to find the DataLoader context in the
+    // global request context
+    resolver.contextToOptions = { [EXPECTED_OPTIONS_KEY]: dataloaderContext };
```
Comment (mickhansen) [inline examples/graphql-yoga/src/server.js]: this can be moved top of file

**Comment by MauriceMahan** [comment]
Very helpful, thanks.

File: examples/graphql-yoga/src/server.js (hunk 1 above)
Comment (mickhansen) [reply examples/graphql-yoga/src/server.js]: Ah actually the line is wrong, it should be `resolver.contextToOptions = {[EXPECTED_OPTIONS_KEY]: EXPECTED_OPTIONS_KEY}`.

contextToOptions just transfers something from the context to the options object.
The map is source:target, it doesn't actually take a value.

**Comment by mikefowler** [comment]
@mickhansen ah, thanks for the clarifications. I amended the commit.

File: examples/graphql-yoga/src/server.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+const server = new GraphQLServer({
+  typeDefs,
+  resolvers,
+  context(req) {
+    // For each request, create a DataLoader context for Sequelize to use
+    const dataloaderContext = createContext(models.sequelize);
+
+    // Tell `graphql-sequelize` where to find the DataLoader context in the
+    // global request context
+    resolver.contextToOptions = contextToOptions;
```
[the remark this answers is not in the document]
Comment (semy) [reply examples/graphql-yoga/src/server.js]: How can I check if the data loader works well?

[the remark this answers is not in the document]
Comment (semy) [reply examples/graphql-yoga/src/server.js]: If I fire:

```
{
  users {
    id
    firstName
    lastName
  }
  
  users2: users {
    id
    firstName
    lastName
  }
}
```

Than I see two sql queries from sequelize:
```
debug: Executing (default): SELECT `id`, `firstName`, `lastName`, `email`, `createdAt`, `updatedAt`, `createdBy`, `updatedBy` FROM `Users` AS `User` ORDER BY `User`.`id` ASC;
debug: Executing (default): SELECT `id`, `firstName`, `lastName`, `email`, `createdAt`, `updatedAt`, `createdBy`, `updatedBy` FROM `Users` AS `User` ORDER BY `User`.`id` ASC;
```

Not one as exprected... My app.js code:

```
// Create Express and Apollo Server
const app = express();
const { sequelize } = db;
const apolloServer = new ApolloServer({
  schema: graphqlSchema,
  formatError: err => logger.error(err),
  context: () => {
    const dataloaderContext = createContext(sequelize);

    return {
      [EXPECTED_OPTIONS_KEY]: dataloaderContext,
    };
  },
});
resolver.contextToOptions = { [EXPECTED_OPTIONS_KEY]: EXPECTED_OPTIONS_KEY };
const apolloOptions = _.merge({ path: '/graphql' }, config.get('graphql'), { app });
apolloServer.applyMiddleware(apolloOptions);
```

And the Schema:

```
const { User } = models;
export default {
  type: new GraphQLList(UserGraphQLType),
  resolve: resolver(User.Model),
};
```

What's wrong? :-(

Thank you for your help :-)

[the remark this answers is not in the document]
Comment (mickhansen) [reply examples/graphql-yoga/src/server.js]: @semy i'm deleting this comment as it does not pertain to the pull request and only acts a distraction. Please ask questions where appropriate.

[the remark this answers is not in the document]
Comment (mickhansen) [reply examples/graphql-yoga/src/server.js]: @mikefowler this one change then i can merge.

**Comment by mikefowler** [comment]
@mickhansen this has been updated again, though it seems like the Travis suite is failing on some specs that should not have been affected by this change.

**Pull request merged**